reference/components/document-manager/extending.rst

```diff
@@ -0,0 +1,18 @@
+Extending the Document Manager
+==============================
+
+Where to put things?
+--------------------
+
+Any classes which relate to the documents or the document manager should
+first, by convention, be placed within a `Document` namespace.
+
+Documents themselves should be placed directly under this namespace and other
+types of class should be placed in sub namespaces with appropriate names. For
+example:
+
+.. code-block:: bash
+
+    src/Bundle/MyBundle/Document/FooDocument.php
+    src/Bundle/MyBundle/Document/Initializer/FooInitializer.php
+    src/Bundle/MyBundle/Document/Subscriber/FooSubscriber.php
```

Comment:
Would be better to do this later as the current practice involves writing directly to the PHPCR node, whereas in the next stable release this will not be necessary.